### PR TITLE
Handle invalid config maxAge type #124

### PR DESCRIPTION
**Pull request by linxea**
- throw type error for invalid maxAge. for example, maxAge: '12345'
- update unit tests

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,7 +61,10 @@ function formatOpts(opts) {
   opts.key = opts.key || 'koa:sess';
 
   // back-compat maxage
-  if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
+  if ('maxAge' in opts) {
+    const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
+    if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
+  } else opts.maxAge = opts.maxage;
```
Comment (dead-horse) [inline index.js]: it does not check opts.maxage here. I think we should keep the original logic, and check opts.maxAge after `if (!('maxAge' in opts)) opts.maxAge = opts.maxage;`

Comment (linxea) [reply index.js]: it's actually using the origin logic? just adding a case for when there is maxAge.

Comment (linxea) [reply index.js]: do you mean you want it this way?

if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
else {
    const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
    if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
}

Comment (dead-horse) [reply index.js]:
```js
if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
```

Comment (dead-horse) [reply index.js]: and I think if we add this validation, somebody's code may be broken and this will be a breaking change

**Pull request closed without merging**